skylark/skydoc.bzl

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Skylark rule for skydoc: a documentation generator tool written in Java."""
+
+load("@bazel_skylib//:skylark_library.bzl", "SkylarkLibraryInfo")
+
+def _skydoc_impl(ctx):
+    """Implementation of the skydoc rule."""
+    out_file = ctx.outputs.out
+    input_files = depset(order = "postorder", direct = [ctx.files.target_file[0]], transitive = [
+        dep[SkylarkLibraryInfo].transitive_srcs
+        for dep in ctx.attr.deps
+    ])
+    args = [
+        str(ctx.files.target_file[0].owner),
```

Comment:
Doc for `owner` says this can be None. Not sure if that happens when the file is a source file, or only when it's a label-less file (which isn't the case here).

Reply:
I'm... not sure, actually. Seems to work fine in this case -- I'm not sure how to make this more robust :)